vector/src/main/java/im/vector/app/features/home/room/list/home/layout/HomeLayoutSettingBottomDialogFragment.kt

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.home.room.list.home.layout
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import dagger.hilt.android.AndroidEntryPoint
+import im.vector.app.R
+import im.vector.app.core.platform.VectorBaseBottomSheetDialogFragment
+import im.vector.app.databinding.BottomSheetHomeLayoutSettingsBinding
+import im.vector.app.features.home.room.list.home.HomeLayoutPreferences
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class HomeLayoutSettingBottomDialogFragment : VectorBaseBottomSheetDialogFragment<BottomSheetHomeLayoutSettingsBinding>() {
+
+    @Inject lateinit var preferences: HomeLayoutPreferences
+
+    override fun getBinding(inflater: LayoutInflater, container: ViewGroup?): BottomSheetHomeLayoutSettingsBinding {
+        return BottomSheetHomeLayoutSettingsBinding.inflate(inflater, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        views.homeLayoutSettingsRecents.isChecked = preferences.areRecentsEnabled()
+        views.homeLayoutSettingsFilters.isChecked = preferences.areFiltersEnabled()
+
+        if (preferences.isAZOrderingEnabled()) {
+            views.homeLayoutSettingsSortName.isChecked = true
+        } else {
+            views.homeLayoutSettingsSortActivity.isChecked = true
+        }
+
+        views.homeLayoutSettingsRecents.setOnCheckedChangeListener { _, isChecked ->
+            preferences.setRecentsEnabled(isChecked)
```

Comment:
I think we shouldn't change preferences before the user clicks to "Done" button. What do you think?

Reply:
@amshakal could you please verify?

Reply:
Yes, I believe no changes should be made before the user hits 'DONE'

Reply:
This is strange to me to have a "DONE" button on a BottomSheet. I would prefer to have the change effect live and visible to the background. For instance when disabling "Show recents", the user will have a direct feedback by seeing the breadcrumbs row vanish in the background. They can even play with the toggle several times. Since it's easy to revert the change, I would simply remove the DONE button. Else what would be done if the user dismiss the bottom sheet without clicking on DONE? Cancel the pending change? Not a good UX to me.

On Android settings and in app settings (preference screen), there is no DONE button, any changes are immediately taking effect. Better to do the same here IMHO.

CC @amshakal

Reply:
Changes are applied when user press DONE, so if user closes dialog in any other way this changes will just be lost. I don't have any strong opinion on this topic. I just felt that seeing changes in ui, while you switch options, feels more interactive, and since here it came with zero costs I decided to implement that way.

TLDR; Both options are good for me and there are no technical issues with both of them, so I let Amsha decide :)

Reply:
While we don't have done button in settings, we can be a bit more explicit with a bottom sheet. I couldn't find anything for or against it in the material guidelines.

I am open to having interactive options but since there is an overlay at the back anyway, the user won't really be able to see the changes properly. 

Let's start with this simple approach and we can iterate on it incase it feels like an odd pattern.

I had a chat with Nikita around this. Benoit, if this feels like a massive blocker/anti pattern. I am more than happy to remove the DONE button.

Reply:
I guess users are used to close bottom sheets either by swiping to the bottom or by clicking on the `back` button.

As a general guidelines, I think that Buttons on BottomSheet are not a good UX pattern. I do not see lots of Button in the example of BottomSheet [here](https://material.io/develop/android/components/bottom-sheet-dialog-fragment) and also in all the apps I use on my phone.

I would definitely remove the DONE button.

Reply:
Got it. In that case Nikita, let's go with benoit's suggestion of removing the CTA. Sorry about the back and forth